src/main/java/org/folio/service/orders/lines/update/instance/WithHoldingOrderLineUpdateInstanceStrategy.java

```diff
@@ -90,81 +128,95 @@ private void removeHoldingUnrecognizedFields(List<JsonObject> holdings) {
   }
 
   private Future<Void> findOrCreateHoldingsAndUpdateItems(OrderLineUpdateInstanceHolder holder,
-      String newInstanceId, RequestContext requestContext) {
-    List<Future<Void>> futures = new ArrayList<>();
-    return retrieveUniqueLocationsAndConsume(holder, requestContext, location -> {
-      String holdingId = location.getHoldingId();
-      futures.add(inventoryHoldingManager.getOrCreateHoldingRecordByInstanceAndLocation(newInstanceId, location, requestContext)
-        .compose(newHoldingId -> {
-          holder.addHoldingRefsToStoragePatchOrderLineRequest(holdingId, newHoldingId);
-          if (ObjectUtils.notEqual(holdingId, newHoldingId)) {
-            return updateItemsHolding(holdingId, newHoldingId, holder.getStoragePoLine().getId(), requestContext);
-          } else {
-            return Future.succeededFuture();
-          }
-        }));
-    })
-      .compose(v -> GenericCompositeFuture.join(new ArrayList<>(futures))
-        .map(ok -> null));
+                                                          String newInstanceId, RequestContext requestContext) {
+    return retrieveUniqueLocations(holder.getStoragePoLine(), requestContext)
+      .compose(locations -> GenericCompositeFuture.join(
+          locations.stream()
+            .map(location -> findOrCreateHoldingsAndUpdateItems(holder, newInstanceId, location, requestContext))
+            .toList()
+        )
+      )
+      .mapEmpty();
+  }
+
+  private Future<Void> findOrCreateHoldingsAndUpdateItems(OrderLineUpdateInstanceHolder holder,
+                                                          String newInstanceId,
+                                                          Location location,
+                                                          RequestContext requestContext) {
+    PoLine poLine = holder.getStoragePoLine();
+    String holdingId = location.getHoldingId();
+    var locationContext = RequestContextUtil.createContextWithNewTenantId(requestContext, location.getTenantId());
```

Comment:
It would be great to remove RequestContextUtils as refactoring old code. I don't see any point to keep class name. Less line more easier to read.
If you are doing, please do it for all other places to be consistent

Reply:
Static imports source of confusion: https://stackoverflow.com/questions/420791/what-is-a-good-use-case-for-static-import-of-methods
Another very good answer here: https://www.reddit.com/r/javahelp/comments/n2jk6e/thoughts_on_static_imports/
```
As a writer, you know where the methods come from. Good for you pal. As a reader, you have no idea - you have to hover a cursor over the method or try to find it another way. It's even harder if you review the code outside of IDE. It simply makes your code less readable because the reader has to know (and continuously keeps a track of) another context(s). 
```

Reply:
Thier reasons make sense. But for popular method, we can still remove Class name I believe. But I am okay with leaving as it was. Thanks